### DISCUSSION TITLE
Moving from Hansson IT Hyper-V VM...

### DISCUSSION THREAD

**Discussion by RJ-Make** [nextcloud/all-in-one#4733] [Questions] [answered]
Hello All, 

I've been using Nextcloud for a few years now via Hansson IT Hyper-V VM on a Windows 10 machine, which is outstanding for those technically challenged.  I'm stuck at NC Version 25 due to the PHP/OS upgrade requirements.

Hansson IT wants ~$800 to update my instance to PHP 8.3 and the latest OS/Nextcloud.  It's just my family who uses NC (5 users) so ~800 bucks is simply too much right now. 

What what are my options?

1) Purchase the latest Hansson IT VM, and reset up everything from scratch. I did this the last time I ran into dependency issues and needless to say it was NOT fun, and to think I'll eventually have to do this again. 

2) Stay with my EOL version of 25. I don't like this ideal as my instance is WAN facing. 

3) Move to AIO (which may still require me to set everything up from scratch), that way when an dependency update is required AIO will take care of this for me. (If that is correct)??

4) ????

Questions:

1) Will AIO handle updating dependencies automatically so I don't find my self in this predicament again in the near future?

2) Can AIO handle access to multiple external Windows folders? (Example:  //192.168.1.26/ServerFolders/CloudData AND //192.168.1.26/ServerFolders/Documents/Tech_Litt) The folders ARE on the same windows machine as docker.

3) Is there a (migration for dummies) way to move all my existing server settings & users from the VM to AIO?

4) Will there be a performance impact moving from Hyper-V to Docker Desktop for Windows?

Thank you in Advance.

**Comment by szaimen** [comment, marked as answer]
Hi,

> 1. Will AIO handle updating dependencies automatically so I don't find my self in this predicament again in the near future?

Yes, AIO takes care of all dependencies. By default, it reminds you to do container updates, there is an option to also do container updates regularly automatically.

> 2. Can AIO handle access to multiple external Windows folders? (Example:  //192.168.1.26/ServerFolders/CloudData AND //192.168.1.26/ServerFolders/Documents/Tech_Litt) The folders ARE on the same windows machine as docker.

If you mount them on the host using fstab inside a subfolder of `/mnt`, you can access all of them by using `--env NEXTCLOUD_MOUNT="/mnt/"`. See https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host.

> 3. Is there a (migration for dummies) way to move all my existing server settings & users from the VM to AIO?

It is not exactly easy but there is this documentation available: https://github.com/nextcloud/all-in-one/blob/main/migration.md

> 4. Will there be a performance impact moving from Hyper-V to Docker Desktop for Windows?

I don't think so but is is easier probably for you to do the migration inside the existing VM and only then thinking about migration over to Docker Desktop for Windows.

**Comment by RJ-Make** [comment]
Thank you so much for your advice. 

1) I've read the provided information on adding Host folder access but how do I add 2 paths?

Add 2 line items?

```
--env NEXTCLOUD_MOUNT="/run/desktop/mnt/host/a/ServerFolders/CloudData/"
--env NEXTCLOUD_MOUNT="/run/desktop/mnt/host/a/ServerFolders/Documents/Tech_Litt/"
```

2) I don't have the skill set to bring my existing VM up to the latest, much less migrate the existing server data/users so I would have to make the move manually and then test.  

As much as this is going to be a challenge for me, it really does look like AIO is the way to go. Even if there is a performance hit, the future dependency update resolution should alone make it worth it.
